relay-server/src/http.rs

```diff
@@ -0,0 +1,299 @@
+                    HttpEncoding::Identity => ContentEncoding::Identity,
+                    HttpEncoding::Deflate => ContentEncoding::Deflate,
+                    HttpEncoding::Gzip => ContentEncoding::Gzip,
+                    HttpEncoding::Br => ContentEncoding::Br,
+                };
+
+                builder.content_encoding(content_encoding);
+            }
+            RequestBuilder::Reqwest {
+                ref mut http_encoding,
+                ..
+            } => {
+                *http_encoding = encoding;
+            }
+        }
+
+        self
+    }
+}
+
+#[allow(clippy::large_enum_variant)]
+pub enum Response {
+    Actix(ClientResponse),
+    Reqwest(reqwest::Response),
+}
+
+impl Response {
+    pub fn status(&self) -> StatusCode {
+        match self {
+            Response::Actix(response) => response.status(),
+            Response::Reqwest(response) => {
+                StatusCode::from_u16(response.status().as_u16()).unwrap()
+            }
+        }
+    }
+
+    pub fn json<T: 'static + DeserializeOwned, E>(
+        self,
+        limit: usize,
+    ) -> Box<dyn Future<Item = T, Error = E>>
+    where
+        E: From<reqwest::Error> + From<actix_web::error::JsonPayloadError> + 'static,
+    {
+        // TODO: apply limit to reqwest
+        match self {
+            Response::Actix(response) => {
+                let future = response.json().limit(limit).map_err(From::from);
+                Box::new(future) as Box<dyn Future<Item = _, Error = _>>
+            }
+            Response::Reqwest(response) => {
+                let future = response.json().boxed_local().compat().map_err(From::from);
+                Box::new(future) as Box<dyn Future<Item = _, Error = _>>
+            }
+        }
+    }
+
+    pub fn consume<E>(self) -> ResponseFuture<Self, E>
+    where
+        E: From<actix_web::error::PayloadError> + 'static,
+    {
+        // consume response bodies to allow connection keep-alive
+        match self {
+            Response::Actix(ref response) => Box::new(
+                response
+                    .payload()
+                    .for_each(|_| Ok(()))
+                    .map(|_| self)
+                    .map_err(From::from),
+            ),
+            Response::Reqwest(_) => {
+                // TODO: is this necessary for reqwest?
+                Box::new(future::ok(self))
+            }
+        }
+    }
+
+    pub fn get_header(&self, key: &str) -> Option<&[u8]> {
+        match self {
+            Response::Actix(response) => Some(response.headers().get(key)?.as_bytes()),
+            Response::Reqwest(response) => Some(response.headers().get(key)?.as_bytes()),
+        }
+    }
+
+    pub fn get_all_headers(&self, key: &str) -> Vec<&[u8]> {
+        match self {
+            Response::Actix(response) => response
+                .headers()
+                .get_all(key)
+                .into_iter()
+                .map(|x| x.as_bytes())
```

Comment:
Let's use speaking names.

```suggestion
                .map(|entry| entry.as_bytes())
```